Sources/starkbank-ecdsa/Utils/Integer.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  File.swift
+//  
+//
+//  Created by Rafael Stark on 2/17/21.
+//
+
+import BigInt
+import Foundation
+
+
+class RandomInteger {
+    
+    /**
+        - Parameter minimum: minimum value of the integer
+        - Parameter maximum: maximum value of the integer
+        - Returns: integer x in the range: minimum <= x <= maximum
+    */
+    public static func between(minimum: BigInt, maximum: BigInt) throws -> BigInt {
+        let result = getBytesNeeded(maximum - minimum)
+        let bytesNeeded = result.0
+        let mask = result.1
+
+        var bytes = [Int](repeating: 0, count: bytesNeeded)
+
+        let status = SecRandomCopyBytes(
+            kSecRandomDefault,
+            bytesNeeded,
+            &bytes
+        )
+
+        var randomValue = BigInt(0)
+
+        if status != errSecSuccess {
+            throw Error.generationError("Error to generate random Int")
+        }
+        
+        (0..<bytesNeeded).forEach {
+            randomValue |= BigInt(bytes[$0]) << (8 * $0)
+        }
+                    
+        randomValue &= mask
+                    
+        if (minimum + randomValue > maximum) {
+//            To improve the security of the generated value, it's better to retry to generate a random value in between the passed range instead of geting it's module.
```

Comment:
// Taking the randomValue module would increase the concentration in a specific region of the range, raising security concerns. To avoid this, we retry the method until the value satisfies the range.